include/bls12-381/scalar.hpp

```diff
@@ -218,18 +218,23 @@ void bn_divn_low(uint64_t *c, uint64_t *d, uint64_t *a, int sa, uint64_t *b, int
 template<size_t N>
 fp fp::modPrime(std::array<uint64_t, N> k)
 {
-    std::array<uint64_t, N> quotient = {0};
-    std::array<uint64_t, N> remainder = {0};
+    static_assert(N >= 6, "input buffer size is smaller than modulus");
+    std::array<uint64_t, N+2> quotient = {0};
+    std::array<uint64_t, N+2> remainder = {0};
     // be conservative with scratch memory (https://github.com/relic-toolkit/relic/blob/ddd1984a76aa9c96a12ebdf5c6786b0ee6a26ef8/src/bn/relic_bn_div.c#L79)
     // with gcc std::array<uint64_t, 6> modulus = fp::MODULUS.d works fine but clang needs the extra words
-    std::array<uint64_t, N> modulus = {0};
+    std::array<uint64_t, N+2> modulus = {0};
     modulus[0] = fp::MODULUS.d[0];
     modulus[1] = fp::MODULUS.d[1];
     modulus[2] = fp::MODULUS.d[2];
     modulus[3] = fp::MODULUS.d[3];
     modulus[4] = fp::MODULUS.d[4];
     modulus[5] = fp::MODULUS.d[5];
-    bn_divn_low(quotient.data(), remainder.data(), k.data(), N, modulus.data(), 6);
+    std::array<uint64_t, N+2> k2 = {0};
+    for (int i=0; i < N; ++i) {
+        k2[i] = k[i];
+    }
+    bn_divn_low(quotient.data(), remainder.data(), k2.data(), N, modulus.data(), 6);
```

Comment:
It sounds to me that more straighforward solution will be to pass N-2 instead and for caller to use modPrime<10> for 8.
In that case we avoid extra array creation & copy.